src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -89,19 +90,21 @@ internal DirectoryLoadContext[] GetDirectoryLoadContextsSnapshot()
             }
         }
 
-        internal void UnloadAll()
+        public void UnloadAll()
         {
-            List<DirectoryLoadContext> contexts;
+            var contexts = ArrayBuilder<DirectoryLoadContext>.GetInstance();
             lock (_guard)
             {
-                contexts = _loadContextByDirectory.Values.ToList();
+                foreach (var (_, context) in _loadContextByDirectory)
+                    contexts.Add(context);
+
                 _loadContextByDirectory.Clear();
             }
 
             foreach (var context in contexts)
-            {
                 context.Unload();
-            }
+
+            contexts.Free();
```

Comment:
This will unload the `AssemblyLoadContext` but it leaves the loader in a bit of a broken / odd state. The loader still has all of it's caches and maps for how assembly paths on disk map to specific properties (name, culture, etc ...). I'm not 100% sure that you could say call `UnloadAll` but then keep using the loader successfully. 

Do we want to consider more of a `Dispose` style method where we tear down the loader?

Reply:
have done this.